ext/vulnsrc/redhat/redhat.go

```diff
@@ -0,0 +1,703 @@
+				if err != nil {
+					// log error and continue
+					log.Error(err)
+					continue
+				}
+				if len(cpeNames) == 0 {
+					log.Warning(fmt.Sprintf("No CPE for: %s %s %s", parsedRmpNvra.Name, parsedRmpNvra.Evr, advisoryDefinition.Metadata.Title))
+				}
+				for _, cpe := range cpeNames {
+					feature.Namespace = database.Namespace{
+						Name:          cpe,
+						VersionFormat: rpm.ParserName,
+					}
+					vulnerability.Affected = append(vulnerability.Affected, feature)
+				}
+			}
+
+		}
+		if len(vulnerability.Affected) > 0 {
+			vulnerabilities = append(vulnerabilities, vulnerability)
+		}
+	}
+	return
+}
+
+// construct the vulnerability name(s) from the given advisory definition
+func ConstructVulnerabilityNames(advisoryDefinition ParsedAdvisory) []string {
+	rhsaName := ParseRhsaName(advisoryDefinition)
+	var vulnNames []string
+	cveNames := ParseCveNames(advisoryDefinition)
+	for _, cveName := range cveNames {
+		vulnNames = append(vulnNames, cveName + " - " + rhsaName)
+	}
+	return vulnNames
+}
+
+// construct the []VulnerabilityID set from the given advisory definition
+func ConstructVulnerabilityIDs(advisoryDefinition ParsedAdvisory) []database.VulnerabilityID {
+	var vulnIDs []database.VulnerabilityID
+	rhsaName := ParseRhsaName(advisoryDefinition)
+	cveNames := ParseCveNames(advisoryDefinition)
+	for _, cveName := range cveNames {
+		vulnID := database.VulnerabilityID{Name: cveName + " - " + rhsaName, Namespace: ParseVulnerabilityNamespace(advisoryDefinition)}
+		vulnIDs = append(vulnIDs, vulnID)
+	}
+	return vulnIDs
+}
+
+// parse the CVE name(s) (e.g.: "CVE-2019-11249") from the given advisory definition
+func ParseCveNames(advisoryDefinition ParsedAdvisory) []string {
+	var cveNames []string
+	for _, cve := range advisoryDefinition.Metadata.Advisory.CveList {
+		cveNames = append(cveNames, cve.Value)
+	}
+	return cveNames
+}
+
+// parse the RHSA name (e.g.: "RHBA-2019:2794") from the given advisory definition
+func ParseRhsaName(advisoryDefinition ParsedAdvisory) string {
+	return strings.TrimSpace(advisoryDefinition.Metadata.Title[:strings.Index(advisoryDefinition.Metadata.Title, ": ")])
+}
+
+// parse the namespace from the given advisory definition
+func ParseVulnerabilityNamespace(advisoryDefinition ParsedAdvisory) string {
+	// use criteria parse result
+	moduleNamespaces := ParseCriteriaForModuleNamespaces(advisoryDefinition.Criteria)
+	if len(moduleNamespaces) > 0 {
+		// use MODULE namespace
+		return moduleNamespaces[0]
+	} else {
+		// use CPE namespace
+		cpeNames, err := ParseCpeNamesFromAffectedCpeList(advisoryDefinition.Metadata.Advisory.AffectedCpeList)
+		if err != nil {
+			// log error and continue
+			log.Error(err)
+			return ""
+		}
+		if len(cpeNames) == 0 {
+			// no namespace found
+			return ""
+		} else {
+			return cpeNames[0]
+		}
+	}
+}
+
+func GetSeverity(sev string) database.Severity {
+	switch strings.Title(sev) {
+	case "None":
+		return database.NegligibleSeverity
```

Comment:
If severity is None please skip the vulnerability and don't store it in database.

Reply:
Added a check to CollectVulnsForAdvisory, which skips processing for any "None" severity advisory.